src/aws/osml/model_runner/inference/feature_utils.py

```diff
@@ -1,20 +1,27 @@
 #  Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
-
+import functools
```

Comment:
Nit: normally we've been leaving a blank line after the copyright.  I'm not picky if we want a blank line or not but we should be consistent.